### PR TITLE
bump-formula-pr: check duplicate PRs as early as possible

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

As discussed in #8011, this PR changes `brew bump-formula-pr` to check for any duplicate PRs as early as possible, i.e. right after determining the formula and the tap in which it's contained.

This allows `brew bump-formula-pr` to exit early and not have to download anything before checking for duplicate PRs.

### Example usages:

When an _open_ duplicate PR is found:

```console
% bump-formula clamav --version 0.102.4
Error: These pull requests may be duplicates:
clamav: add libtool dependency https://github.com/Homebrew/homebrew-core/pull/57734
Duplicate PRs should not be opened. Use --force to override this error.
```

When a _closed_ duplicate PR is found:

```console
% bump-formula convox --version 3.0.32
Error: These pull requests may be duplicates:
convox 3.0.32 https://github.com/Homebrew/homebrew-core/pull/58044
Duplicate PRs should not be opened. Use --force to override this error.
```

-----

### Other misc. changes:

* Fixed a bug introduced in #8004 (my bad!) where the `version` stanza isn't updated if all three of `--version/--url/--sha256` are specified (or alternatively `--version/--tag/--revision`). The specific patch that fixes this is:

  ```patch
  -    forced_version = false
  +    forced_version = new_version.present?
  ```

* Replaced all occurrences of `backup_file` with the more accurate name `old_contents` (as suggested in #8011)

### CODE REVIEW

**Comment by SeekingMeaning** [comment]
Still need to test the newest commit, but it passes `brew style` at least

File: Library/Homebrew/dev-cmd/bump-formula-pr.rb [hunk 1]
```diff
@@ -155,15 +159,16 @@ def bump_formula_pr
       new_url.sub "mirrors.ocf.berkeley.edu/debian", "mirrorservice.org/sites/ftp.debian.org/debian"
     end
     new_mirrors ||= [new_mirror] unless new_mirror.nil?
-    new_version = args.version
     old_url = formula_spec.url
     old_tag = formula_spec.specs[:tag]
     old_formula_version = formula_version(formula, requested_spec)
     old_version = old_formula_version.to_s
-    forced_version = false
+    forced_version = !new_version.nil?
```
Comment (jonchang) [inline Library/Homebrew/dev-cmd/bump-formula-pr.rb:166]:
```suggestion
    forced_version = new_version.present?
```
We have activesupport so this should work and is clearer than a double negation.

Comment (SeekingMeaning) [reply Library/Homebrew/dev-cmd/bump-formula-pr.rb:166]: Oh that's much better, see anything else that needs changing?

**Comment by jonchang** [comment]
Thanks @SeekingMeaning!

**Pull request merged**